controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -271,6 +282,37 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	crdPredicate := predicate.Funcs{
+		CreateFunc: func(e event.TypedCreateEvent[client.Object]) bool {
+			if e.Object.GetName() == ClusterClaimCrdName && !r.AvailableCrds[ClusterClaimCrdName] {
+				r.Log.Info("CustomResourceDefinition %s was Created.", e.Object.GetName())
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.TypedDeleteEvent[client.Object]) bool {
+			if e.Object.GetName() == ClusterClaimCrdName && !r.AvailableCrds[ClusterClaimCrdName] {
```

Comment:
I believe you have a bug here
```suggestion
			if e.Object.GetName() == ClusterClaimCrdName && r.AvailableCrds[ClusterClaimCrdName] {
```